pandas/tests/indexes/test_base.py

```diff
@@ -1099,13 +1108,48 @@ def test_get_indexer_nearest(self):
             tm.assert_numpy_array_equal(actual, np.array(expected,
                                                          dtype=np.intp))
 
+            actual = idx.get_indexer([0.2, 1.8, 8.5], method=method,
+                                     tolerance=[1.5, 1, 1])
+            tm.assert_numpy_array_equal(actual, np.array(expected,
+                                                         dtype=np.intp))
+
         for method, expected in zip(all_methods, [[0, -1, -1], [-1, 2, -1],
                                                   [0, 2, -1]]):
             actual = idx.get_indexer([0.2, 1.8, 8.5], method=method,
                                      tolerance=0.2)
             tm.assert_numpy_array_equal(actual, np.array(expected,
                                                          dtype=np.intp))
 
+            actual = idx.get_indexer([0.2, 1.8, 8.5], method=method,
+                                     tolerance=[0.5, 0.2, 0.2])
+            tm.assert_numpy_array_equal(actual, np.array(expected,
+                                                         dtype=np.intp))
```

Comment:
can u use parametrize here to make the tests more readable

Reply:
can be a separate test

Reply:
I'm a little of unsure of what to parameterize here: in each for loop, I actually just copy the last test (`actual = idx.(...`, 'tm.assert_numpy_array_equal(...`) using scalar tolerance but change the tolerance to an array. I could add some comments to make it clearer?

Reply:
parametrize on method,expected